ansible_collection/hpe/nimble/plugins/modules/hpe_nimble_protection_schedule.py

```diff
@@ -0,0 +1,510 @@
+#!/usr/bin/python
+
+# Copyright 2020 Hewlett Packard Enterprise Development LP
+#
+# Licensed under the Apache License, Version 2.0 (the "License"); you may not use this
+# file except in compliance with the License. You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software distributed
+# under the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS
+# OF ANY KIND, either express or implied. See the License for the specific
+# language governing permissions and limitations under the License.
+
+# author alok ranjan (alok.ranjan2@hpe.com)
+
+ANSIBLE_METADATA = {'metadata_version': '1.1',
+                    'status': ['preview'],
+                    'supported_by': 'community'}
+
+DOCUMENTATION = r'''
+---
+author:
+  - Alok Ranjan (@ranjanal)
+description: Manage protection schedule on HPE Nimble Storage group.
+module: hpe_nimble_protection_schedule
+options:
+  at_time:
+    required: False
+    type: int
+    default: 0
+    description:
+    - Time of day when snapshot should be taken. In case repeat frequency specifies more than one snapshot
+      in a day then the until_time option specifies until what time of day to take snapshots.
+  change_name:
+    required: False
+    type: str
+    description:
+    - Change the name of existing protection schedule.
+  days:
+    required: False
+    type: str
+    default: 'all'
+    description:
+    - Specifies which days snapshots should be taken. Comma separated list of days of the week or 'all'.
+  description:
+    required: False
+    type: str
+    description:
+    - Description of the schedule.
+  disable_appsync:
+    required: False
+    type: bool
+    default: False
+    description:
+    - Disables application synchronized snapshots and creates crash consistent snapshots instead.
+  downstream_partner:
+    required: False
+    type: str
+    description:
+    - Specifies the partner name if snapshots created by this schedule should be replicated.
+  name:
+    required: True
+    type: str
+    default: None
+    description:
+    - Name of snapshot schedule to create.
+  num_retain:
+    required: False
+    type: int
+    description:
+    - Number of snapshots to retain. If replication is enabled on this schedule the array will always
+      retain the latest replicated snapshot, which may exceed the specified retention value. This is necessary to ensure efficient replication performance.
```

Comment:
Maybe a new line so this line doesn't go to far.

Reply:
our flake config is set to 160 per lines. i have refactored it.